client/components/relatedWidget/RelatedCarousel.jsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import ScrollMenu from 'react-horizontal-scrolling-menu';
+import ProductList from './ProductList.jsx';
+import ArrowButton from './ArrowButton.jsx';
+import axios from 'axios';
+
+const selected = null;
+const ArrowLeft = ArrowButton({
+  text: '<',
+  className: 'arrow-prev'
+});
+const ArrowRight = ArrowButton({
+  text: '>',
+  className: 'arrow-next'
+});
+
+class RelatedCarousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selected: selected,
+      relatedIds: [],
+      relatedProducts: [],
+      relatedImages: []
+    };
+    this.productItems = ProductList(
+      {
+        products: this.state.relatedProducts,
+        images: this.state.relatedImages
+      },
+      this.state.selected,
+      this.props.getNewProduct);
+    this.onSelect = this.onSelect.bind(this);
+  }
+
+  onSelect(key) {
+    this.setState({ selected: key });
+  }
+
+  getRelatedProduct(productId) {
+    axios.get(`/api/fec2/hrnyc/products/${productId}/related`)
+    .then(results=>{
+      this.setState({relatedIds: results.data});
+      let products = results.data.map(relatedId=>{
+        return axios.get(`/api/fec2/hrnyc/products/${relatedId}`);
+      });
+      let images = results.data.map(relatedId=>{
+        return axios.get(`/api/fec2/hrnyc/products/${relatedId}/styles`);
+      });
+      return {
+        products: products,
+        images: images
+      };
+    })
+    .then(results=>{
+      // console.log('unresolved results:', results);
+          Promise.all(results.products)
+          .then(results=>{
+            return results.map((res)=>{
+              return res.data;
+            });
+          })
+          .then(results=>{ this.setState({relatedProducts: results}); })
+          .catch(err=>{ console.log(err); });
+
+          Promise.all(results.images)
+          .then(results=>{
+            return results.map((res)=>{
+              return res.data.results[0].photos[0].thumbnail_url;
+            });
+          })
+          .then(results=>{ this.setState({relatedImages: results}); })
+          .catch(err=>{ console.log(err); });
+    })
+    .catch(err=>{ console.log(err); });
+  }
+
+  buildCarousel() {
+    this.productItems = ProductList(
+      {
+        products: this.state.relatedProducts,
+        images: this.state.relatedImages
+      },
+      this.state.selected,
+      this.props.getNewProduct);
+      // console.log('inside buildCarousel');
+  }
+
+  componentDidMount() {
+    // console.log('componentDidMount');
+    // console.log(this.props.product)
+    this.buildCarousel();
+  }
```

Comment:
It looks like buildCarousel does the same thing as the initial state of `this.productItems`
If that is true then you immediately replace `this.productItems` when `componentDidMount()` runs.  Any reason you can't just set the initial value of `this.productItems` to `[]` and that way remove the duplicate logic and the extra calculations?